src/sentry/integrations/slack/unfurl/__init__.py

```diff
@@ -1,14 +1,14 @@
 from __future__ import annotations
 
 import enum
-from typing import Any, Callable, Mapping, NamedTuple, Pattern
+from typing import Any, Callable, Mapping, NamedTuple, Optional, Pattern
 
 from django.http.request import HttpRequest
 
 from sentry.models import Integration, User
 
 UnfurledUrl = Mapping[Any, Any]
-ArgsMapper = Callable[[str, Mapping[str, str]], Mapping[str, Any]]
+ArgsMapper = Callable[[str, Mapping[str, Optional[str]]], Mapping[str, Any]]
```

Comment:
this is most of the rest of the changes -- the `args` from django are actually `None`-able so the `Mapper`s need to handle that